rafthttp/http.go

```diff
@@ -271,6 +273,19 @@ func (h *streamHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// put known peers in header in case follower joins a new leader
+	// and can't resolve leader's URL since it can't receive a ConfChange yet
+	peers := []string{}
+	for id, p := range h.peerGetter.list() {
+		var urls []string
+		for _, url := range p.urls() {
+			urls = append(urls, url.String())
+		}
+		peers = append(peers, fmt.Sprintf("%x=%s", uint64(id), strings.Join(urls, ",")))
+	}
+	// peerId0=peerURL,...;peerID1=peerURL,...;...
+	w.Header().Set("X-Raft-Peers", strings.Join(peers, ";"))
```

Comment:
The follower doesn't know the local peer address, so it wouldn't be able to connect to get the header.